code/game/mecha/mecha.dm

```diff
@@ -1137,6 +1137,9 @@ obj/mecha/proc/can_use(mob/user)
 			if(mmi.brainmob)
 				occupant.loc = mmi
 			mmi.mecha = null
+			if(istype(mmi,/obj/item/device/mmi/posibrain))

```

Comment:
You can accomplish this by giving MMIs an ```update_icon()``` and overriding it on Posibrains.
(MMIs handle their icons in attackby, also with an ```istype()``` for alien brains, so that needs a cleanup regardless)